node/external/blockAPI/baseBlock.go

```diff
@@ -668,3 +719,152 @@ func proofToAPIProof(proof data.HeaderProofHandler) *api.HeaderProof {
 		IsStartOfEpoch:      proof.GetIsStartOfEpoch(),
 	}
 }
+
+func (bap *baseAPIBlockProcessor) addMbsAndNumTxsAsyncExecutionBasedOnExecutionResult(apiBlock *api.Block, blockHeader data.HeaderHandler, headerHash []byte, options api.BlockQueryOptions) error {
```

Comment:
not very clear about the naming here, can we have async execution not based on execution result?